cmd/integrationArtifactUpload_test.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type integrationArtifactUploadMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newIntegrationArtifactUploadTestsUtils() integrationArtifactUploadMockUtils {
+	utils := integrationArtifactUploadMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunIntegrationArtifactUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Create Test", func(t *testing.T) {
+
+		tempDir, tmpErr := ioutil.TempDir("", "")
+		defer os.RemoveAll(tempDir) // clean up
+		assert.NoError(t, tmpErr, "Error when creating temp dir")
+		files := &mock.FilesMock{}
+		path := filepath.Join(tempDir, "iflow4.zip")
+		files.AddFile(path, []byte("dummy content"))
+		createErr := ioutil.WriteFile(path, []byte("dummy content"), 0755)
+		assert.NoError(t, createErr)
+		assert.True(t, files.HasFile(path))
+
+		config := integrationArtifactUploadOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowName:    "flow4",
+			IntegrationFlowID:      "flow4",
+			IntegrationFlowVersion: "1.0.4",
+			PackageID:              "CICD",
+			FilePath:               path,
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "", ResponseBody: ``, TestType: "PositiveAndCreateIntegrationDesigntimeArtifactResBody"}
+
+		err := runIntegrationArtifactUpload(&config, nil, &httpClient)
+
+		if assert.NoError(t, err) {
+
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "https://demo/api/v1/IntegrationDesigntimeArtifactSaveAsVersion?Id='flow4'&SaveAsVersion='1.0.4'", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "POST", httpClient.Method)
+			})
+		}
+	})
+
+	t.Run("Successfull Integration Flow Update Test", func(t *testing.T) {
+
+		tempDir, tmpErr := ioutil.TempDir("", "")
```

Comment:
Instead of working with a tempory directory the file mocking facilities can be used.